custom_components/mbapi2020/switch.py

```diff
@@ -6,159 +6,77 @@
-                    str(e),
-                )
-
-    async_add_entities(entities)
-
-
-class SwitchTurnOn(Protocol):
-    """Protocol for a callable that asynchronously turns on a MercedesME switch."""
-
-    async def __call__(self, **kwargs) -> None:
-        """Asynchronously turn on the switch."""
-
-
-class SwitchTurnOff(Protocol):
-    """Protocol for a callable that asynchronously turns off a MercedesME switch."""
-
-    async def __call__(self, **kwargs) -> None:
-        """Asynchronously turn off the switch."""
-
-
-class SwitchIsOn(Protocol):
-    """Protocol for a callable that checks if a MercedesME switch is on."""
-
-    def __call__(self) -> bool:
-        """Check if the switch is currently on."""
-
-
-@dataclass(frozen=True)
-class MercedesMeSwitchEntityConfig(MercedesMeEntityConfig):
+@dataclass(frozen=True, kw_only=True)
+class MercedesMeSwitchEntityDescription(SwitchEntityDescription):
     """Configuration class for MercedesMe switch entities."""
 
-    turn_on: SwitchTurnOn | None = None
-    turn_off: SwitchTurnOff | None = None
-    is_on: SwitchIsOn | None = None
-
-    def __post_init__(self):
-        """Post-initialization checks to ensure required fields are set."""
-        if self.capability_check is None:
-            raise ValueError(f"capability_check is required for {self.__class__.__name__}")
-        if self.turn_on is None:
-            raise ValueError(f"turn_on is required for {self.__class__.__name__}")
-        if self.turn_off is None:
-            raise ValueError(f"turn_off is required for {self.__class__.__name__}")
-
-    def __repr__(self) -> str:
-        """Return a string representation of the MercedesMeSwitchEntityConfig instance."""
-        return (
-            f"{self.__class__.__name__}("
-            f"internal_name={self.id!r}, "
-            f"entity_name={self.entity_name!r}, "
-            f"feature_name={self.feature_name!r}, "
-            f"object_name={self.object_name!r}, "
-            f"attribute_name={self.attribute_name!r}, "
-            f"capability_check={self.capability_check!r}, "
-            f"attributes={self.attributes!r}, "
-            f"device_class={self.device_class!r}, "
-            f"icon={self.icon!r}, "
-            f"entity_category={self.entity_category!r}, "
-            f"turn_on={self.turn_on!r}, "
-            f"turn_off={self.turn_off!r}, "
-            f"is_on={self.is_on!r})"
-        )
+    attributes: list[str] | None = None
+    is_on_fn: Callable[[MercedesMESwitch], Callable[[], Coroutine[Any, Any, bool]]]
+    turn_on_fn: Callable[[MercedesMESwitch], Callable[[], Coroutine[Any, Any, None]]]
+    turn_off_fn: Callable[[MercedesMESwitch], Callable[[], Coroutine[Any, Any, None]]]
+    check_capability_fn: Callable[[Car], Callable[[], Coroutine[Any, Any, bool]]]
 
+SWITCH_CONFIGS: list[MercedesMeSwitchEntityDescription] = [
+    MercedesMeSwitchEntityDescription(
+        key="precond",
+        translation_key="Pre-entry climate control",
```

Comment:
Translation key does not follow the rule r"^(?!.+[_-]{2})(?![_-])[a-z0-9-_]+(?<![_-])$" . You can remove this here as the key is has the correct format.